lab-michaelm/modules/list.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+
+const List = module.exports = function() {
+  this.length = 0;
+};
+
+List.prototype.push = function(callback) {
+  this[this.length] = callback;
+  this.length++;
+};
+
+List.prototype.pop = function() {
+  if (this.length < 1) return console.error('There were no items to remove.');
+  let result = this[this.length - 1];
+  delete this[this.length - 1];
+  this.length--;
+  return result;
+};
+
+List.prototype.map = function(callback) {
+  let newList = new List();
+  for (var i = 1; i < this.length; i++) {
+    () => newList.push(callback(this[i]));
```

Comment:
Are we calling a function to call a function here? It seems like we should just be able to call 'newlist.push(etc..)'. That could be the cause of getting undefined here.